emily_cron/app/services/deposit_processor.py

```diff
@@ -16,6 +16,56 @@
 class DepositProcessor:
     """Service for processing deposits."""
 
+    def process_rbf_transactions(
+        self,
+        enriched_deposits: list[EnrichedDepositInfo],
+    ) -> list[DepositUpdate]:
+        """Process RBF transactions.
+        Args:
+            enriched_deposits: List of enriched deposit information
+        Returns:
+            list[DepositUpdate]: List of deposit updates
+        """
+        updates = []
+
+        # Find transactions with RBF replacements
+        rbf_txs = [tx for tx in enriched_deposits if tx.rbf_txids]
+        if not rbf_txs:
+            return updates
+
+        logger.info(f"Found {len(rbf_txs)} transactions with RBF replacements")
+
+        # Group by replacement chain
+        rbf_groups = self._group_rbf_transactions(rbf_txs)
+
+        # Process each group
+        for group_txids in rbf_groups.values():
+            # Find all transactions in this group
+            group_txs = [tx for tx in enriched_deposits if tx.bitcoin_txid in group_txids]
+
+            # Check if any transaction in this group is confirmed
+            confirmed_txs = [tx for tx in group_txs if tx.confirmed_height is not None]
+
+            if not confirmed_txs:
+                logger.debug(f"No confirmed transactions found for group {group_txids}")
+                continue
+
+            # If we have confirmed transactions, mark all unconfirmed ones as FAILED
+            for tx in group_txs:
+                if tx.confirmed_height is None:
+                    logger.info(
+                        f"Marking RBF'd transaction {tx.bitcoin_txid} as FAILED (replaced by confirmed tx)"
+                    )
+                    updates.append(
+                        DepositUpdate(
+                            bitcoin_txid=tx.bitcoin_txid,
+                            bitcoin_tx_output_index=tx.bitcoin_tx_output_index,
+                            status=RequestStatus.FAILED.value,
+                            status_message=f"Replaced by confirmed tx {confirmed_txs[0].bitcoin_txid}",
```

Comment:
We should consider adding the replacement transaction ID to the deposit's event object instead of putting it within the status message. That way it will be easier to work with programmatically.